src/develop/develop.c

```diff
@@ -1206,17 +1212,20 @@ void dt_dev_write_history_ext(dt_develop_t *dev, const int imgid)
   sqlite3_step(stmt);
   sqlite3_finalize(stmt);
   GList *history = dev->history;
+  // the history size is NOT defined by current history_end but by the number actually written
```

Comment:
Ah, did miss that and this is the issue. The history_end must be always kept as-is. The user can set an old history and keep it as long as he wants until the history is compressed. So this part is probably to be reverted.